app/src/main/java/com/example/composetemplate/repositories/LoginRepository.kt

```diff
@@ -2,68 +2,110 @@ package com.example.composetemplate.repositories
         if (user.email.isEmpty().not()) {
-            firebaseDataSource.auth.signInWithEmailAndPassword(user.email, password)
-                .addOnCompleteListener { task ->
-                    if (task.isSuccessful) {
-                        firebaseDataSource.createOrUpdateUser(user, loginCallback)
-                    }  else {
-                        loginCallback(null, task.exception)
-                    }
+            authDataSource.signInWithEmailAndPassword(user, password) { createdUser, exception ->
+                if (createdUser != null && exception == null)
+                    createOrUpdateUser(user, loginCallback)
+                else {
+                    loginCallback(null, exception)
                 }
+            }
         } else
             loginCallback(null, Exception("email is empty"))
     }
 
-    /**
-     * Create or update in database.
-     * We can change the request to other server anytime we want id we change the service we use.
-     * */
-    fun createOrUpdateUserRequest(user: User): BaseRequest {
-        return firebaseDataSource.createOrUpdateUserRequestForFirebase(user)
-    }
+    fun logOut() = authDataSource.logout()
 
-    // First fetch the access token
-    fun getUserAccessToken(successCallback: SuccessCallback) = firebaseDataSource.getUser(successCallback)
+    fun resetPassword(email: String, successCallback: SuccessCallback) = authDataSource.resetPassword(email,successCallback)
 
-    fun getUserRequest() = firebaseDataSource.getUserRequest()
+    override fun getUser(loginCallback: LoginCallback) {
+        authDataSource.getUser { user, exception ->
+            if (exception == null) {
+                ioScope.launch {
+                    val baseRequest = createBaseRequest(RequestType.GET_USER, authDataSource, user) ?: return@launch
+                    val httpResponse = (networkManager.sendRequest(baseRequest) as? HttpResponse)
+                    httpResponse?.let { response ->
+                        val remoteUser = response.body<User>()
+                        loginCallback(remoteUser,null)
+                    } ?: run {
+                        loginCallback(null,Exception())// check exception
+                    }
+                }
+            } else loginCallback(null,exception)
+        }
+    }
 
-    //logOut user in firebase
-    fun logOut() = firebaseDataSource.logOut()
+    override fun createOrUpdateUser(user: User, loginCallback: LoginCallback) {
+        ioScope.launch {
+            val baseRequest = createBaseRequest(RequestType.CREATE_USER, authDataSource, user) ?: return@launch
+            val httpResponse = (networkManager.sendRequest(baseRequest) as? HttpResponse)
+            val success = httpResponse?.isSuccessful() == true
+            if (success) loginCallback(user, null) else loginCallback(null, Exception())
+        }
+    }
 
+    /** This is base request to to use in our network call architecture.
```

Comment:
We need to settle on a consistent format for comments. Personally, I prefer the /* */ style, but I’ve noticed that you've used // comments in several places. Let’s decide to use the /* */ style moving forward for consistency.